pieceio/pieceio.go

```diff
@@ -26,20 +21,19 @@ type CarIO interface {
 }
 
 type pieceIO struct {
-	padReader PadReader
-	carIO     CarIO
-	store     filestore.FileStore
-	bs        blockstore.Blockstore
+	carIO CarIO
+	store filestore.FileStore
+	bs    blockstore.Blockstore
 }
 
-func NewPieceIO(padReader PadReader, carIO CarIO, store filestore.FileStore, bs blockstore.Blockstore) PieceIO {
-	return &pieceIO{padReader, carIO, store, bs}
+func NewPieceIO(carIO CarIO, store filestore.FileStore, bs blockstore.Blockstore) PieceIO {
+	return &pieceIO{carIO, store, bs}
 }
 
-func (pio *pieceIO) GeneratePieceCommitment(payloadCid cid.Cid, selector ipld.Node) ([]byte, filestore.File, error) {
+func (pio *pieceIO) GeneratePieceCommitment(payloadCid cid.Cid, selector ipld.Node) ([]byte, filestore.File, uint64, error) {
```

Comment:
You're writing the CAR to a tempfile here. Some Linux distros (like [Arch](https://wiki.archlinux.org/index.php/Tmpfs)) use a ramdisk for `/tmp`. Does `CreateTemp()` write to `/tmp`? If so, I would expect this to be the source of OOMs for miners committing large sectors. 

A potential solution would be to modify `pieceIO#GeneratePieceCommitment` to return a `Reader` instead of a `File`. You could use (`os.Pipe`)[https://golang.org/pkg/os/#Pipe], passing the write side of the pipe to `pio.carIO.WriteCar` in a goroutine and return the read side of the pipe.

Reply:
golang's temp file is not writing to a ram disk -- it puts it in the directory you specify

Reply:
I will look at piping.